daemon/api_snapshots.go

```diff
@@ -137,3 +142,49 @@ func changeSnapshots(c *Command, r *http.Request, user *auth.UserState) Response
 
 	return AsyncResponse(nil, &Meta{Change: chg.ID()})
 }
+
+type snapshotExportResponse struct {
+	SetID      uint64
+	ExportSize uint64
+}
+
+func (s snapshotExportResponse) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Length", strconv.FormatUint(s.ExportSize, 10))
+
+	snapshotExport(context.TODO(), uint64(s.SetID), w)
+}
+
+type countingOnlyWriter struct {
```

Comment:
We have sizer (sizer.go) in snapshotstate/backend that does the same thing. Probably not for this PR, but it would be nice to share it between the two (move to *util package?).

Reply:
Added https://github.com/snapcore/snapd/pull/9014 and if that is accepted I will replace the exiting copies with this one.